core/lib/spree/service_module.rb

```diff
@@ -23,6 +23,20 @@ def failure?
       end
     end
 
+    ResultError = Struct.new(:value) do
+      def to_s
+        if value&.respond_to?(:full_messages)
+          value.full_messages.join(', ')
+        else
+          value&.to_s
```

Comment:
This will produce an empty string if the value will be nil. I think it should return nil.